nibabel/streamlines/compact_list.py

```diff
@@ -0,0 +1,232 @@
+import numpy as np
+
+
+class CompactList(object):
+    """ Class for compacting list of ndarrays with matching shape except for
+    the first dimension.
+    """
+
+    BUFFER_SIZE = 87382*4  # About 4 Mb if item shape is 3 (e.g. 3D points).
+
+    def __init__(self, iterable=None):
+        """
+        Parameters
+        ----------
+        iterable : iterable (optional)
```

Comment:
Could also say more about what the elements in the iterable can be.  For example, I think they can be array-like (lists, tuples etc).